app/actions/CommonActions.scala

```diff
@@ -45,6 +45,12 @@ object CommonActions {
 
   val CachedAction = resultModifier(Cached(_))
 
+  val StoreAcquisitionDataAction = new ActionBuilder[Request] {
+    def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]) = block(request).map(result => {
+      request.getQueryString("acquisitionData").fold(result)(a => result.withSession(request.session.data.toSeq ++ Seq("acquisitionData" -> a) :_*))
```

Comment:
a should be  `acquisitionDataFromQueryString` I guess?
this line is quite long and may be good to split into a val or def or something just to get names on some of these things

Reply:
so here it's only added into the session once the request has completed, so it won't be in the session during the current request.  is that correct?

Reply:
yep that's correct!